crypto/src/ed25519.rs

```diff
@@ -339,7 +341,7 @@ impl KeyPair for Ed25519KeyPair {
     fn copy(&self) -> Self {
         Self {
             name: Ed25519PublicKey::from_bytes(self.name.as_ref()).unwrap(),
-            secret: Ed25519PrivateKey::from_bytes(self.name.as_ref()).unwrap(),
+            secret: Ed25519PrivateKey::from_bytes(self.secret.as_ref()).unwrap(),
```

Comment:
Yes ! There seems to be a test missing in there.

Reply:
You're absolutely correct! Although Tasos has kindly added it in another PR.